index.js

```diff
@@ -926,6 +926,82 @@ PersonaClient.prototype.getProfileByGuid = function(opts, callback){
     });
     req.end();
 };
+
+/**
+ * Get all profiles for an array of GUIDs.
+ *
+ * TODO Neither the client lib nor Persona impose restriction on amount of GUIDs requested. Limit in calling app for now.
+ *
+ * @param  {object}   opts     [description]
+ * @param  {array}    opts.guids  Array of GUIDs to fetch profiles for
+ * @param  {string}   opts.token  Auth token
+ * @param  {string}   opts.xRequestId Optional request ID to pass through in logging.
+ * @param  {Function} callback
+ */
+PersonaClient.prototype.getProfilesForGuids = function getProfilesForGuids(opts, callback) {
+    validateOpts(opts,{ guids: _.isArray, token: _.isString });
+    var guids = opts.guids;
+    var token = opts.token;
+    var xRequestId = opts.xRequestId || uuid.v4();
+    var _this = this;
+
+    var ids = '';
+
+    // generate a comma-sep list of user's from the list of users we need to process
+    guids.forEach(function (guid) {
+        ids += (ids !== '') ? ',' + guid : guid;
+    });
+
+    var options = {
+        hostname: _this.config.persona_host,
+        port: _this.config.persona_port,
+        path: '/users?guids=' + ids,
+        method: 'GET',
+        headers: {
+        'Authorization': 'Bearer ' + token,
+        'User-Agent': _this.userAgent,
+        'X-Request-Id': xRequestId
+        }
+    };
+
+    var personaReq = _this.http.request(options, function personaReq(personaResp) {
+        var userString = '';
+
+        personaResp.on('data', function onData(chunk) {
+            userString += chunk;
+        });
+
+        personaResp.on('end', function onEnd() {
+            if (personaResp.statusCode === 200) {
+                var data = JSON.parse(userString);
+                var results = [];
+                if (!_.isEmpty(data)) {
+                    if (_.isArray(data)) {
+                        results = data;
+                    } else {
+                        results.push(data);
+                    }
+                }
+                callback(null, results);
+            } else {
+                var error = new Error();
+                error.http_code = personaResp.statusCode || 404;
```

Comment:
If there is no response code then 0 should be returned. This is how a lot of libraries behave (such as jQuery).